api/src/test/java/org/pmiops/workbench/db/service/UserRecentResourceServiceTest.java

```diff
@@ -0,0 +1,147 @@
+package org.pmiops.workbench.db.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.UserRecentResourceDao;
+import org.pmiops.workbench.db.dao.UserRecentResourceServiceImpl;
+import org.pmiops.workbench.db.dao.UserDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.Cohort;
+import org.pmiops.workbench.db.model.UserRecentResource;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.test.Providers;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.liquibase.LiquibaseAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.sql.Timestamp;
+import java.util.Date;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@Import(LiquibaseAutoConfiguration.class)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class UserRecentResourceServiceTest {
+
+  UserRecentResourceServiceImpl userRecentResourceService;
+
+  @Autowired
+  UserDao userDao;
+  @Autowired
+  WorkspaceDao workspaceDao;
+  @Autowired
+  CohortDao cohortDao;
+  @Autowired
+  UserRecentResourceDao notebookCohortCacheDao;
+  @Mock
```

Comment:
opt: you can define a `TestConfiguration` in this file and provide an instance of WorkbenchConfig to avoid using a mock here. Then you can also just Autowire in your target service instead of constructing it yourself.

Example: https://github.com/all-of-us/workbench/blob/7b992ab8fc1ccaf604326833aedafc77069a7c2d/api/src/test/java/org/pmiops/workbench/api/OfflineClusterControllerTest.java#L67